### DISCUSSION TITLE
How to see the Python version for greptime-windows-amd64-pyo3-v0.5.0

### DISCUSSION THREAD

**Discussion by amythos** [GreptimeTeam/greptimedb#3117] [Q&A] [answered]
How to see the Python version for greptime-windows-amd64-pyo3-v0.5.0

**Comment by tisonkun** [comment, marked as answer]
I created https://github.com/GreptimeTeam/greptimedb/pull/3118 to print the version in INFO level.

I'm not quite familiar with pyo3, but on my environment it's 3.11.6 the same as my platform's Python.

I don't know if it just uses the system-preinstalled Python (I guess it is) or bundle one. Perhaps @messense can provide more information.

To @messense, the question is what Python is in used when `Python::with_gil` is called after `pyo3::prepare_freethreaded_python()` is initialized previously in a Rust binary process.

**Reply by tisonkun** [reply]
On the very last version, when you use RustPython backend:

```
Initialized RustPython interpreter 3.11.0alpha (heads/main:9ed513741, Apr 17 2023, 22:03:55) 
```

... and when you use PyO3 backend:

```
Started CPython Interpreter 3.11.6 (main, Nov  2 2023, 04:39:43) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

The PyO3 version depends on your system-wise installed Python interpreter while the RustPython one is bundled. #3118 adds two INFO level logs to print this information.
